daemon/prune.go

```diff
@@ -150,3 +154,72 @@ func (daemon *Daemon) ImagesPrune(config *types.ImagesPruneConfig) (*types.Image
 
 	return rep, nil
 }
+
+// localNetworksPrune removes unused local networks
+func (daemon *Daemon) localNetworksPrune(config *types.NetworksPruneConfig) (*types.NetworksPruneReport, error) {
+	rep := &types.NetworksPruneReport{}
+	var err error
+	// When the function returns true, the walk will stop.
+	l := func(nw libnetwork.Network) bool {
+		nwName := nw.Name()
+		predefined := runconfig.IsPreDefinedNetwork(nwName)
+		if !predefined && len(nw.Endpoints()) == 0 {
```

Comment:
I understand you want to use`daemon.DeleteNetwork()` for removing the network instead of just calling `nw.Delete()` because the the former will also log the event.

But it also run the `isPredefinedNetwork()` and `isdynamic` check. I am thinking you may want to drop the `!predefined` check in this function and skip the warn log in case the error returned by `daemon.Deletenetwork()` is of type `NewRequestForbiddenError`.

Something like:
```
	l := func(nw libnetwork.Network) bool {
		if len(nw.Endpoints()) != 0 {
			return false
		}
		err := daemon.DeleteNetwork(nw.ID())
		if err == nil {
			rep.NetworksDeleted = append(rep.NetworksDeleted, nwName)
		} else if _, ok := err.(NewRequestForbiddenError); !ok {
			logrus.Warnf("could not remove network %s: %v", nw.Name(), err)
		}
		return false
	}
```